src/Domain/FunctionsOfTime/FixedSpeedCubic.cpp

```diff
@@ -0,0 +1,92 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/FunctionsOfTime/FixedSpeedCubic.hpp"
+
+#include <cmath>
+#include <memory>
+
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+
+namespace domain::FunctionsOfTime {
+FixedSpeedCubic::FixedSpeedCubic(const double initial_function_value,
+                                 const double initial_time,
+                                 const double velocity,
+                                 const double decay_timescale) noexcept
+    : initial_function_value_(initial_function_value),
+      initial_time_(initial_time),
+      velocity_(velocity),
+      squared_decay_timescale_(square(decay_timescale)) {}
+
+std::unique_ptr<FunctionOfTime> FixedSpeedCubic::get_clone() const noexcept {
+  return std::make_unique<FixedSpeedCubic>(*this);
+}
+
+template <size_t MaxDerivReturned>
+std::array<DataVector, MaxDerivReturned + 1> FixedSpeedCubic::func_and_derivs(
+    const double t) const noexcept {
+  static_assert(MaxDerivReturned < 3, "The maximum available derivative is 2.");
+
+  // initialize result for the number of derivs requested
+  std::array<DataVector, MaxDerivReturned + 1> result =
+      make_array<MaxDerivReturned + 1>(DataVector(1, 0.0));
+
+  const double dt = t - initial_time_;
+  const double denom = squared_decay_timescale_ + square(dt);
+  ASSERT(fabs(denom) > 0.0,
+         "FixedSpeedCubic denominator should not be zero; if evaluating at "
+         "t == t0, then do not set the decay timescale tau to 0.");
```

Comment:
[optional] If tau=0 the function is just `f_0 + v dt`, so you could return `(f_0, v, 0)`.